### PR TITLE
fix: legacy deployments [BB-6237]

### PR DESCRIPTION
**Pull request by Agrendalath**
This includes several fixes and QoL improvements to present and legacy deployment scripts. Please check it commit-by-commit to get the complete picture.

### CODE REVIEW

File: modules/openedx/s3/buckets.tf [hunk 1]
```diff
@@ -1,33 +1,52 @@
...
   provider = aws.s3
 }
 
-resource "aws_s3_bucket" "edxapp_course_discovery" {
-  count = var.enable_course_discovery ? 1 : 0
-  bucket = lower(join("-", [var.client_shortname, "edxapp", var.environment, "course-discovery"]))
-  acl = "private"
+resource "aws_s3_bucket_acl" "edxapp_course_discovery_acl" {
+  count  = length(aws_s3_bucket.edxapp_course_discovery)
+  bucket = aws_s3_bucket.edxapp_course_discovery[0].id
```
Comment (farhaanbukhsh) [inline modules/openedx/s3/buckets.tf:36]: What will happen when `count    = var.enable_course_discovery ? 1 : 0` count = 0, this statement will throw error right?

**Review by farhaanbukhsh — APPROVED** [review]
👍 

- ✅ I did not test this but the code has been used to manage infra of client
- ✅  I read through the code
- ❌  I checked for accessibility issues
- ✅   Includes documentation
- ❌  I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.

File: modules/openedx/s3/buckets.tf (hunk 1 above)
Comment (Agrendalath) [reply modules/openedx/s3/buckets.tf:36]: @farhaanbukhsh, why would it fail? We're checking the length one line above as a condition to create/destroy this resource. 

Besides, I've deployed this with `var.enable_course_discovery = false`.

Comment (farhaanbukhsh) [reply modules/openedx/s3/buckets.tf:36]: Ahh! my bad :)

**Pull request merged**